packages/@aws-cdk/cdk/lib/cloudformation/resource.ts

```diff
@@ -229,6 +232,29 @@ export class Resource extends Referenceable {
   }
 }
 
+export class TaggableResource extends Resource implements ITaggable {
+  /**
+   * TagManager to manage the propagation and assignment of tags
+   */
+  public readonly tags: TagManager;
+
+  constructor(parent: Construct, name: string, props: ResourceProps) {
+    super(parent, name, props);
+    // const initialTags = !!props.properties ? props.properties.tags || {} : {};
```

Comment:
Why not? And then you don't have to do the merge in `toCloudFormation`?

Reply:
funny thing I missed deleting this ... 

So I started here, but where it blows up is the `props.tags` type supports just a `cdk.Token` so where this broke was when a developer creates a L1 and passes a function or value token to the tags.

```
const tagsArray: cloudformation.Tag[] = [];
new cloudformation.L1(parent, 'Name', { tags: new cdk.Token(() => tagsArray)});
tagsArray.push({key: 'Key', value: someOtherL1.ref});
```

In that situation the initial tags are lost because `TagManager` doesn't support the array of tokens or just a plain token itself. I could go back and change that implementation to support it, but I'm not sure how many things are going to unravel. I think now that TagManager is a Token itself it may be do-able.

Reply:
I see... well... technically, a token can also resolve to a CloudFormation intrinsic, so even `resolve`ing it at synthesis time won't really help you:

```
new cloudformation.L1(parent, 'Name', { tags: new FnImportValue(...) )});
```

To be honest, I really don't see much value in being able to pass in a token for `tags` at the L1 layer. Why don't we special case this in cfn2ts so that Tags will always use concrete types (remember, textual tokens can be stringified and assigned to keys/values, so we are not prohibiting this use case). This way, you won't have that issue and you could process the initial tags in the ctor.

Reply:
I see the special case for tags in cfn2ts. Are you saying we change from this today:

```
        tags?: Array<cdk.Tag | cdk.Token> | cdk.Token;
```
to
```
        tags?: Array<cdk.Tag | cdk.Token>;
```
**OR** this is what I was thinking
```
        tags?: Array<cdk.Tag>;
```
Which one were you thinking?

Reply:
I think `cdk.Tag[]` is sufficient. Keep it simple. I don't see a strong use case for any of the other options.

The only use case where tokens might be needed for tags is as tag values, which are `string`s, and we know how to represent tokens as strings :-)